src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -413,7 +413,7 @@ export async function run(
     };
     const checkBatch = await submitBatch(t, shaderBuilder, shaderBuilderParams, pipelineCache);
     checkBatch();
-    void t.queue.onSubmittedWorkDone().finally(batchFinishedCallback);
+    void t.queue.onSubmittedWorkDone();
```

Comment:
Calling onSubmittedWorkDone without using the result doesn't do anything (though chrome has a bug where some actions will not flush, and this will flush them)

Do you mean to `await` it here? I think that would work.

Reply:
> Do you mean to await it here?

Yes. Fixed and retested and it still works.